app/youtube/service_test.go

```diff
@@ -111,6 +111,79 @@ func TestService_Do(t *testing.T) {
 	assert.Equal(t, "/tmp/648f79b3a05ececb8a37600aa0aee332f0374e01.mp3", duration.FileCalls()[3].Fname)
 }
 
+// nolint:dupl // test if very similar to TestService_RSSFeed
+func TestService_DoIsRelevantFilter(t *testing.T) {
+
+	chans := &mocks.ChannelServiceMock{
+		GetFunc: func(ctx context.Context, chanID string, feedType ytfeed.Type) ([]ytfeed.Entry, error) {
+			return []ytfeed.Entry{
+				{ChannelID: chanID, VideoID: "vid1", Title: "Prefix1: title1", Published: time.Now()},
+				{ChannelID: chanID, VideoID: "vid2", Title: "Prefix2: title2", Published: time.Now()},
+				{ChannelID: chanID, VideoID: "vid3", Title: "Prefix2: title3", Published: time.Now()},
+			}, nil
+		},
+	}
+	downloader := &mocks.DownloaderServiceMock{
+		GetFunc: func(ctx context.Context, id string, fname string) (string, error) {
+			return "/tmp/" + fname + ".mp3", nil
+		},
+	}
+
+	duration := &mocks.DurationServiceMock{
+		FileFunc: func(fname string) int {
+			return 1234
+		},
+	}
+
+	tmpfile := filepath.Join(os.TempDir(), "test.db")
+	defer os.Remove(tmpfile)
+
+	db, err := bolt.Open(tmpfile, 0o600, &bolt.Options{Timeout: 1 * time.Second})
+	require.NoError(t, err)
+	boltStore := &store.BoltDB{DB: db}
+	svc := Service{
+		Feeds: []FeedInfo{
+			{ID: "channel1", Name: "name1", Type: ytfeed.FTChannel, Filter: FeedFilter{Include: "Prefix2", Exclude: "title3"}},
```

Comment:
I'd suggest adding a case with some regex filters. Maybe for channel2 in addition to the current one